doc/semiboolmat.xml

```diff
@@ -0,0 +1,175 @@
+      </Item>
+      <Item>
+        <C>D[1][2]</C>, <C>D[i][i]</C>, for <M>i \in \{2, \ldots, n\}</M>,
+        and <C>D[n][1]</C>.
+      </Item>
+    </List>
+    This monoid has nearly <C>2 ^ (n ^ 2)</C> elements. 
+  </Description>
+</ManSection>
+<#/GAPDoc>
+
+<#GAPDoc Label="ReflexiveBooleanMatMonoid">
+   <ManSection>
+    <Oper Name="ReflexiveBooleanMatMonoid" Arg="d"/>
+    <Returns>A monoid of boolean matrices.</Returns>
+    <Description>
+      If <A>d</A> is a positive integer less than or equal to <C>5</C>, then
+      this operation returns the monoid consisting of all reflexive <A>d</A> by
+      <A>d</A> boolean matrices. A boolean matrix <C>mat</C> is <E>reflexive</E>
+      if each entry of its leading diagonal is <K>true</K>, i.e. if
+      <C>mat[i][i]</C> is <K>true</K> for all <M>i \in \{1, \ldots, d\}</M>.
+      <P/>
+
+      The generating sets for the monoids returned by
+      <C>ReflexiveBooleanMatMonoid</C> are pre-computed, and read from a file.
+      Small generating sets are not known for <M><A>d</A> \geq 6</M>.
+
+      <Example><![CDATA[
+gap> S := ReflexiveBooleanMatMonoid(4);
+<monoid of 4x4 boolean matrices with 38 generators>
+gap> Size(S);
+4096]]></Example>
+    </Description>
+  </ManSection>
+<#/GAPDoc>
+
+<#GAPDoc Label="HallMonoid">
+   <ManSection>
+    <Oper Name="HallMonoid" Arg="d"/>
+    <Returns>A monoid of boolean matrices.</Returns>
+    <Description>
+      If <A>d</A> is a positive integer less than or equal to <C>5</C>, then
+      this operation returns the monoid consisting Hall matrices of degree
+      <A>d</A>. A <E>Hall matrix</E> is a boolean matrix in which every column
+      and every row contains at least one <K>true</K> entry.  Equivalently, a
+      Hall matrix is a boolean matrix than contains a permutation. <P/>
+
+      A Hall matrix of dimension <A>d</A> corresponds to a solution to Hall's
+      Marriage Problem, when there are <A>d</A> men and <A>d</A> women.  Thus
+      the number of solutions to Hall's Marriage Problem in this instance is the
+      number of elements of <C>HallMonoid(<A>d</A>)</C>. <P/>
+
+      The operation <C>HallMonoid</C> returns a monoid with a generating set
+      that is minimal in size. These generating sets are pre-computed, and a
+      minimal generating set is not known for larger dimensions.
+
+      <Example><![CDATA[
+gap> S := HallMonoid(3);
+<monoid of 3x3 boolean matrices with 4 generators>
+gap> Size(S);
+247]]></Example>
+    </Description>
+  </ManSection>
+<#/GAPDoc>
+
+<#GAPDoc Label="GossipMonoid">
+   <ManSection>
+    <Oper Name="GossipMonoid" Arg="d"/>
+    <Returns>A monoid of boolean matrices.</Returns>
+    <Description>
+      If <A>d</A> is a positive integer, then this operation returns the
+      <A>d</A> by <A>d</A> gossip monoid. The <E>gossip monoid</E> is the
+      monoid generated by the collection of all <A>d</A> by <A>d</A> boolean
+      matrices that define an equivalence relation. <P/>
```

Comment:
Reference `IsEquivalenceBooleanMat` and maybe be more specific about what the actual generating set is in GAP.